packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -8,16 +8,64 @@
 // @remove-on-eject-end
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const crypto = require('crypto');
 const errorOverlayMiddleware = require('react-dev-utils/errorOverlayMiddleware');
 const evalSourceMapMiddleware = require('react-dev-utils/evalSourceMapMiddleware');
 const noopServiceWorkerMiddleware = require('react-dev-utils/noopServiceWorkerMiddleware');
 const ignoredFiles = require('react-dev-utils/ignoredFiles');
 const paths = require('./paths');
-const fs = require('fs');
 
-const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
 const host = process.env.HOST || '0.0.0.0';
 
+// Get the https config
+// Return cert files if provided in env, otherwise just true or false
+function httpsConfig() {
+  const { SSL_CRT_FILE, SSL_KEY_FILE, HTTPS } = process.env;
+  const https = HTTPS === 'true';
+
+  if (https && SSL_CRT_FILE && SSL_KEY_FILE) {
+    const cwd = process.cwd();
+    const crtFile = path.resolve(cwd, SSL_CRT_FILE);
+    const keyFile = path.resolve(cwd, SSL_KEY_FILE);
+    if (!fs.existsSync(crtFile)) {
+      throw new Error(
+        `You specified ${chalk.cyan(
+          'SSL_CRT_FILE'
+        )} in your env, but the file "${chalk.yellow(crtFile)}" doesn't exist.`
+      );
+    }
+    if (!fs.existsSync(keyFile)) {
+      throw new Error(
+        `You specified ${chalk.cyan(
+          'SSL_KEY_FILE'
+        )} in your env, but the file "${chalk.yellow(keyFile)}" doesn't exist.`
+      );
+    }
+    const config = {
+      key: fs.readFileSync(keyFile),
+      cert: fs.readFileSync(crtFile),
+    };
+    try {
+      crypto.publicEncrypt(config.cert, new Buffer(''));
```

Comment:
I was wondering, why do you do this, I mean the encrypted value is not used anywhere?

Reply:
@joggienl Thanks for reviewing. The reason for this is to try and stay consistent with create react app being easy to use and debug. This check, and the one below is to validate the certs passed in. If they are invalid it logs an easy to understand error. Without this the standard error message is some random text related to the crypto package, and it isn't clear what went wrong.

Reply:
@alexbrazier thanks for the quick reply!
You're correct about the more easy to understand error! That is actually much appreciated :-)

You did not answer what I was trying to get to know though, why is the call to `crypto.publicEncrypt` made? Apologies for the unclear question!

Based on your reply I actually think the assumption I had beforehand was correct: by encrypting the `Buffer` with the public key (and below, the private one) you actually "validate" the certificate or key.

I do have a suggestion, and that is to put a comment to explain a bit what happens, not only from user point of view. Unless you are using the encrypted value this code actually does something else then it reads, it validates the certificate. This can easily be unclear when looking to at a later moment.

Reply:
@joggienl yeah, if the `encrypt` throws then it means that the key/cert was invalid. I will update the PR to add a comment as you suggest

Reply:
I've updated it with comments, and moved the https config into a separate file